.eslintrc.cjs

```diff
@@ -36,6 +36,7 @@ module.exports = {
         additionalHooks: "useDebouncedEffect",
       },
     ],
+    "react/no-object-type-as-default-prop": "error",
```

Comment:
[See doc](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-object-type-as-default-prop.md)

Primitive default values for props are OK but object/array props are not unless you define them outside of the definition of the function component.

A few times I have made this mistake, which leads to excessive re-renders and is hard to track down. I see no downsides to adding this rule.